### PR TITLE
Test ert slurm driver with ad-hoc cluster

### LINKED ISSUE #6296: Add integration test to run polycase on SLURM
Implement poly example using slurm queue, which can be done via `ubuntu-latest` at first.

### PR DESCRIPTION
**Pull request by berland**
**Issue**
Resolves #6296

**Approach**
Install the Slurm software on the github action runner, declare the action runner host as a compute node, and run poly example. 

## Pre review checklist

- [x] Read through the code changes carefully after finishing work
- [ ] Make sure tests pass locally (after every commit!)
- [x] Prepare changes in small commits for more convenient review (optional)
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Updated documentation
- [ ] Ensured that unit tests are added for all new behavior (See 
    [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)),
    and changes to existing code have good test coverage.

## Pre merge checklist
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: .github/workflows/test_ert_with_slurm.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      run: |
+        find . -name "*.whl" -exec pip install "{}[dev]" \;
+
+    - name: Install and setup slurm
+      run: |
+        set -e
+
+        sudo apt install slurmd slurmctld -y
+
+        cat <<EOF > slurm.conf
```
Comment (xjules) [inline .github/workflows/test_ert_with_slurm.yml:43]: Is this not missing a bracket ? `>`

Comment (berland) [reply .github/workflows/test_ert_with_slurm.yml:43]: No, this is a bash "HERE" document; 

https://en.wikipedia.org/wiki/Here_document

File: .github/workflows/test_ert_with_slurm.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        # Self-configure the node:
+        slurmd -C | grep NodeName >> slurm.conf
+
+        cat slurm.conf
+
+        sudo mv slurm.conf /etc/slurm/
+        sudo systemctl start slurmd
+        sudo systemctl start slurmctld
+
+        sinfo
```
Comment (xjules) [inline .github/workflows/test_ert_with_slurm.yml:61]: Could you maybe put some comments into the code for those who don't know what does it entail to startup slurm? (like me). Thanks!

File: .github/workflows/build_and_test.yml [hunk 3]
```diff
@@ -118,6 +118,13 @@ jobs:
       python-version: ${{ matrix.python-version }}
       test-type: ${{ matrix.test-type }}
 
+  test-slurm:
+    needs: [build-linux]
+    uses: ./.github/workflows/test_ert_with_slurm.yml
+    with:
+      os: ubuntu-latest
+      python-version: '3.10'
```
Comment (xjules) [inline .github/workflows/build_and_test.yml:126]: it does not work with Python 3.8?

Comment (berland) [reply .github/workflows/build_and_test.yml:126]: It probably does, I was just thinking of saving resources..

File: .github/workflows/test_ert_with_slurm.yml (hunk 2 above)
Comment (berland) [reply .github/workflows/test_ert_with_slurm.yml:62]: Fixed

File: .github/workflows/test_ert_with_slurm.yml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+jobs:
+  test-poly-on-slurm:
+    name: Run ert tests
+    timeout-minutes: 20
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
```
Comment (kwinkunks) [inline .github/workflows/test_ert_with_slurm.yml:17]: This fetches all branches and all tags, maybe `1` is sufficient?

Comment (berland) [reply .github/workflows/test_ert_with_slurm.yml:17]: yes, and `1` is even the default.

Comment (berland) [reply .github/workflows/test_ert_with_slurm.yml]: fixed

**Review by xjules — APPROVED** [review]
LGTM! :rocket:

**Pull request merged**